usagov/usagov-2021
# USAGOV-1771: Redirect for /libraries

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1771

## Description

Adds a redirect from /libraries to /libraries-and-archives in the nginx configuration

## Type of Changes

  - Add screenshot showing what it should look like
- [x] Other - nginx redirect config (requires "bin/init") 

## Testing Instructions



### Change Requirements



### Validation Steps
This redirect is for the `www` app, which we don't run locally. You can test a copy of it with the CMS (but it is a hassle so you can also trust that I did this if you want): 
1. In .docker/src-cms/etc/nginx/partials/cms.conf.tmpl, uncomment line 17. 
2. Run `docker compose down; docker system prune -a`, then `bin/init`. (This is the hassle) 
3. Make sure there is not already an alias for /libraries in the URL Redirects in drupal. Check http://localhost/admin/config/search/redirect?redirect_source__path=libraries&redirect_redirect__uri=&status_code=All&language=All and delete the alias if it's present.
4. Run `wget http://localhost/libraries`. You're looking for a few lines indicating a redirect; they will look like this: 
```
--2024-07-26 12:41:17--  http://localhost/libraries
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: http://localhost/libraries-and-archives [following]
--2024-07-26 12:41:17--  http://localhost/libraries-and-archives
```

Why not test with a browser? I found that my web browser was hanging on to a renaming of /libraries to /Libraries, which must be built-in to Drupal core and which doesn't carry over to the static site.

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
6. Find the commit of this pull request.
7. Build and deploy the changes.
8. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
